editor/filesystem_dock.cpp

```diff
@@ -1729,12 +1729,12 @@ void FileSystemDock::_folder_removed(String p_folder) {
 
 void FileSystemDock::_rename_operation_confirm() {
 	String new_name;
-	TreeItem *s = tree->get_selected();
-	int col_index = tree->get_selected_column();
+	TreeItem *s = tree->get_edited();
+	int col_index = tree->get_edited_column();
 
-	if (tree->has_focus()) {
+	if (s != nullptr) {
```

Comment:
```suggestion
	if (s) {
```
`s` is a pointer (integer), so it can be used as boolean (with null being 0/false) and that's what we do more often. Although either way is fine.